guardian/french-thrift
# Update to Apache Thrift v0.21.0

## What does this change?

- Merges in from tag 0.21.0 from the upstream [apache/thrift](https://github.com/apache/thrift/tree/0.21.0) repository to bring french-thrift up to date with the latest version of Apache Thrift
- No new patches from our side, but existing patches from the mobile-apps branch are maintained (such as the fixes needed to provide async support, e.g. https://github.com/guardian/french-thrift/commit/eb925d452286b8bdc6911cb633826938ab54423e and https://github.com/guardian/french-thrift/commit/94d343bc584fe38ef0c2617538b1906a529480f9). 

I followed the directions outlined in the [GitHub documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-branch-from-the-command-line) for syncing a fork with the upstream repository from the command line.

Hopefully this will resolve the security vulnerabilities that are currently being reported in this project: 

![image](https://github.com/user-attachments/assets/68b572b3-2c74-49d3-b7c2-ea3afbf4ac17)

A few GitHub Actions are failing on this PR. I'm not too worried about that, because I can see that the same actions are failing for the same reasons on the `master` branch of the Apache Thrift library too (I think they were introduced with [this](https://github.com/apache/thrift/commit/e9f63e0bbdc9316f57dd2458c27ff18ed1df60fe#diff-f7c0ee7ade1356a3c568cd5c6542fb816601f63854263b7f9e04aa4ee4944ca2) commit). I've followed the testing steps below to check these changes. 

## How to test

- [x] Before merging, create a prerelease from this branch: DONE [here](https://github.com/guardian/french-thrift/releases/tag/v0.21.0-prerelease)
- [x] Create a branch in [Bridget](https://github.com/guardian/bridget) to check this prerelease works: DONE [here](https://github.com/guardian/bridget/pull/178), and the [`validate-thrift`](https://github.com/guardian/bridget/actions/runs/12158640356/job/33907017150?pr=178) action succeeded.
  - Set the [validate-thrift](https://github.com/guardian/bridget/blob/main/.github/actions/validate-thrift/Dockerfile#L3) action's Dockerfile to use the prerelease 
  - Disable the [paths filter](https://github.com/guardian/bridget/blob/main/.github/workflows/validate-thrift.yml#L8) for the validate-thrift workflow so it runs for the test pull request

An example of this testing flow from the last Thrift update is here: https://github.com/guardian/bridget/compare/main...pair/test-update-thrift

## Release process
This repository doesn't include a release process as such. Once this PR is approved and merged to the mobile-apps branch, and provided that the vulnerabilities are indeed resolved by this update, create a Release in the GitHub Release UI with tag v0.21.0-gu1.

This new release can then can be used by [Guardian Thrift Swift](https://github.com/guardian/thrift-swift/), [Ophan Thrift Swift](https://github.com/guardian/ophan-thrift-swift) and [Bridget](https://github.com/guardian/bridget).
